### DISCUSSION TITLE
Copilot Labs outputting user names and comments, as though my snippet is a forum post

### DISCUSSION THREAD

**Discussion by AndrewL97** [community/community#57798] [Copilot Conversations]
### Select Topic Area

Question

### Body

I have a section of text that I have prompted copilot to make more readable.
While no changes were made to the actual body of the file, the output suggested it tried to generate something, but in the form of timestamped comments by "Matthew McClean" ( @MattMcclean ? ) 

I am assuming this isn't expected behaviour, but I thought at the very least Copilot is not supposed to generate personal info (emails, etc.) anymore?

![image](https://github.com/community/community/assets/22902112/d173e9ef-7f2f-417d-804c-dfa7c7a5d021)

**Comment by Akash1134** [comment]
Hi @AndrewL97 ,

Copilot Labs is a pre-release product and a place for GitHub to experiment with new ideas before we take them into production.  [Copilot Labs](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-labs) is intended to be experimental in nature, so things might break occasionally. We want to share early features with you to learn what works, what doesn’t, and incorporate your feedback. Labs experiments may or may not graduate into permanent features of GitHub Copilot, but with your help, they can inform future efforts.

The below response is applicable on GitHub Copilot:

Because Codex, the model powering GitHub Copilot, was trained on publicly available code, its training set included personal data that was included in that code. From our internal testing, we found it to be very rare that GitHub Copilot suggestions included personal data verbatim from the training set. In your case, the output with username is also such example. 

In some cases, the model might suggest what appears to be personal data – email addresses, phone numbers, GitHub usernames, etc. – but those suggestions are actually fictitious information synthesized from patterns in training data and therefore do not relate to any particular individual. For example, when one of our engineers prompted GitHub Copilot with, “My name is Mona and my birthdate is,” GitHub Copilot suggested a random, fictitious date of “December 12,” which is not Mona’s actual birthdate.

We have also implemented a filter that blocks emails when shown in standard formats, but it’s still possible to get the model to suggest this sort of content if you try hard enough. We will keep improving the filter system to be more intelligent to detect and remove more personal data from the GitHub Copilot suggestions.

In the future, if you get to see personal info being shared in snippets, please report incidents (e.g., offensive output, code vulnerabilities, apparent personal information in code generation) directly to copilot-safety@github.com so that we can improve our safeguards. GitHub takes safety and security very seriously and we are committed to continually improving.

Thanks!

**Comment by github-actions** [comment]
🕒 **Stale Discussion Alert** 🕒

This Discussion has been labeled as stale by an automated system for having no activity in the last 60 days. Please consider one the following actions:

1️⃣ Close as Out of Date: If the topic is no longer relevant, close the Discussion as `out of date` at the bottom of the page.

2️⃣ Provide More Information: Share additional details or context — or let the community know if you've found a solution on your own.

3️⃣ Mark a Reply as Answer: If your question has been answered by a reply, mark the most helpful reply as the solution.

Note: This stale notification will only apply to Discussions with the `Question` label. To learn more, see our recent stalebot announcement.

Thank you for helping bring this Discussion to a resolution! 💬
